ui/src/FindBar.tsx

```diff
@@ -92,12 +133,59 @@ export class FindBar extends React.PureComponent<{}, FindBarState> {
     this.dispatchEventToPdfJs("find");
   }
 
-  handleNextButtonClick() {
-    this.dispatchEventToPdfJs("findagain");
+  wrapIndex(index: number, len: number) {
+    return (index % len + len) % len;
   }
 
-  handlePreviousButtonClick() {
-    this.dispatchEventToPdfJs("findagain", true);
+  selectSymbol(id: string, boxId: string) {
+    this.context.setSelectedEntity(id, "symbol");
+    this.context.setSelectedAnnotationId(convertToAnnotationId(id));
+    this.context.setSelectedAnnotationSpanId(boxId);
+  }
+
+  unselectSymbol() {
+    this.context.setSelectedEntity(null, null);
+    this.context.setSelectedAnnotationId(null);
+    this.context.setSelectedAnnotationSpanId(null);
+  }
+
+  moveToNextSymbol(movement: number) {
+    const { matches, mappingToBounds, selectedSymbol } = this.props;
+    const newEntityId = matches[
+      this.wrapIndex(matches.indexOf(selectedSymbol || '') + movement, matches.length)
+    ];
+   
+    const newBoxId = mappingToBounds.has(newEntityId) ? mappingToBounds.get(newEntityId).id : '';
+    this.selectSymbol(newEntityId, newBoxId);
+  }
+
+  handleNextButtonClick(e) {
+    if (this.state.mode === 'symbol-search') {
+      this.moveToNextSymbol(1);
+    } else {
+      this.dispatchEventToPdfJs("findagain");
+    }
+    e.preventDefault();
+  }
+
+  handlePreviousButtonClick(e) {
+    if (this.state.mode === 'symbol-search') {
+      this.moveToNextSymbol(-1);
+    } else {
+      this.dispatchEventToPdfJs("findagain", true);
+    }
+    e.preventDefault();
+  }
+
+  closeFinder = (e) => {
+    this.setState(initialState);
+    this.unselectSymbol();
+    // TODO: fix this super hacky way to remove the green highlight on close.
```

Comment:
Hahaha got anything in mind for a better implementation for this one?

Reply:
If not, you can still leave the TODO in here and assign it to me (`TODO(andrewhead)`) as I might get a chance to figure out pdf.js's preferred way of terminating searches when I clean up our integration with pdf.js's search capabilities.

Reply:
Hahah no didn't have anything in particular, I figured if we keep this implementation then I would go rummaging through pdf.js for a better API :)